SimPEG/electromagnetics/frequency_domain/simulation.py

```diff
@@ -239,6 +239,85 @@ def Jtvec(self, m, v, f=None):
                 Jtv += np.real(df_dmT_sum)
 
         return mkvc(Jtv)
+    
+    def getJ(self, m, f=None):
```

Comment:
Would there be an issue putting this in the getter of Jmatrix instead?

Reply:
It shouldn't be and issue so can and will do!

Reply:
Is there a point to having a Jmatrix property? J is dependent on the passes model, so it makes sense that it is still a function.

Reply:
@jcapriot So just store an hidden attribute `_Jmatrix`?

Reply:
But then where would you pass the model?

what your proposing would always require you to do:
```
sim.model = m
sim.Jmatrix @ x
```

Reply:
Well Jmatrix always requires fields, and model gets set in the `fields` call. A bit hidden, but that's the common path for all inversions.

Reply:
I think the common path for all the inversions in SimPEG is to evaluate and stash fields for a given model, then pass those to dpred, Jvec and JTvec.

The general form of those functions is to use a fields object if given, or else compute fields given the model then use them and discard the fields.